src/test/java/com/iexec/worker/PingServiceTests.java

```diff
@@ -24,29 +25,143 @@
 import org.mockito.InjectMocks;
 import org.mockito.Mock;
 import org.mockito.MockitoAnnotations;
+import org.mockito.Spy;
 
+import java.util.concurrent.TimeUnit;
+
+import static org.assertj.core.api.Assertions.assertThat;
 import static org.mockito.Mockito.*;
 
 public class PingServiceTests {
 
+    private final static String PING_THREAD_NAME = "ping-1";
     public static final String SESSION_ID = "SESSION_ID";
     public static final String OTHER_SESSION_ID = "OTHER_SESSION_ID";
 
-    @InjectMocks
-    private PingService pingService;
     @Mock
     private CustomCoreFeignClient customCoreFeignClient;
     @Mock
     private CoreConfigurationService coreConfigurationService;
     @Mock
     private WorkerService workerService;
 
+    @Spy
+    @InjectMocks
+    private PingService pingService;
+
     @Before
     public void beforeEach() {
         MockitoAnnotations.openMocks(this);
 
     }
 
+    @Test
+    public void shouldRunPingAsynchronouslyWhenTriggeredOneTime() throws InterruptedException {
+        ThreadNameWrapper threadNameWrapper = new ThreadNameWrapper();
+        String mainThreadName = Thread.currentThread().getName();
+        doAnswer(invocation -> TestUtils.saveThreadNameThenCallRealMethod(threadNameWrapper, invocation))
+                .when(pingService).pingScheduler();
+        
+        pingService.triggerSchedulerPing();
+        TimeUnit.MILLISECONDS.sleep(10);
+        // Make sure pingScheduler() method is called 1 time
+        verify(pingService).pingScheduler();
+        // Make sure pingScheduler() method is executed 1 time
```

Comment:
`ping` instead of `pingScheduler`? There are a few copies of that line throughout the file, I'll let you track and erase them all ;)